lint_conandata.py

```diff
@@ -40,26 +52,33 @@ def check_alternative_archives(url:str, orig_size: int | None):
     else:
         # Not an archive
         return
+
     results = []
+    if "/-/archive/" in url:
+        # This is most likely a GitLab archive, can limit the check to just .tar.bz2
+        archive_suffixes = [".tar.bz2"]
+        results.append((orig_size, url))
+
     for suffix in archive_suffixes:
         new_url = without_suffix + suffix
         if new_url == url:
-            if orig_size is None:
-                # Server does not report sizes and only worse formats remain
-                return
-            # Skip the request for the original URL
-            continue
-        response = test_url(new_url, timeout=2)
-        if response and response.ok:
-            if "Content-Length" not in response.headers:
-                print(f"a potentially smaller archive exists at {new_url}")
-                return
-            size = int(response.headers["Content-Length"])
-            results.append((size, new_url))
-    results = sorted(results)
-    if results:
+            size = orig_size
+        else:
+            response = test_url(new_url, timeout=2)
+            if not response or not response.ok:
+                continue
+            size = _get_content_length(response)
+        results.append((size, new_url))
+        if size is None:
+            break
+
+    if any(size is None for size, _ in results):
         best_size, best_url = results[0]
-        if orig_size and best_size and best_size < orig_size:
+        if best_url != url:
+            print(f"a potentially smaller archive exists at {best_url}")
+    else:
+        best_size, best_url = min(results)
+        if best_url != url:
             improvement = (orig_size - best_size) / orig_size
```

Comment:
```suggestion
            assert orig_size is not None, f"orig_size is None for {url=}, {results=}, {archive_suffixes=}"
            improvement = (orig_size - best_size) / orig_size
```
I think with the current code, this assert cannot be triggered, but I still prefer checking.
This will silence [prospector](https://github.com/ericLemanissier/conan-center-lint-conandata/actions/runs/6962190397/job/18945379183#step:5:21) and be a safeguard in the future, if there is a change in the code above.